### PR TITLE
Add auxiliary semantic segmentation output to segmentation predict

### PR DESCRIPTION
**Pull request by tedo001**
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Adds semantic segmentation logits support to segmentation inference, while also tightening a keypoint augmentation edge case and improving distributed training behavior. 🚀

### 📊 Key Changes
- Added optional `semseg` storage to `Results` so segmentation predictions can carry auxiliary semantic segmentation logits.
- Updated the segmentation predictor to extract optional semseg outputs from the model head and attach per-image semseg tensors to each `Results` object. 🧩
- Modified the segmentation head and proto block so inference can return both mask prototypes and optional semseg outputs when available.
- Relaxed the proto block condition from training-only to availability-based semseg output, enabling inference-time semseg prediction.
- Fixed a `v8_transforms()` edge case by guarding `flip_idx` validation when `kpt_shape` is `None`, preventing failures on non-pose configs. 🛠️
- Updated DDP setup to use `broadcast_buffers=False`, which may reduce unnecessary buffer synchronization in multi-GPU training.

### 🎯 Purpose & Impact
- Enables richer segmentation inference workflows by exposing auxiliary semseg logits directly in prediction results. 🎯
- Improves flexibility for downstream consumers that need both instance masks and semantic segmentation outputs from one forward pass.
- Prevents an augmentation-related validation error for datasets without keypoint metadata, improving robustness.
- May improve distributed training stability or efficiency in setups where broadcasting buffers is unnecessary.
- Main user-facing impact is in segmentation prediction behavior, with low-risk supporting changes in training and data augmentation.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @tedo001, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help with review readiness—an Ultralytics engineer will assist you soon. For this `Feat/semseg predict` update, please review the checklist below to help ensure smooth integration. 😊

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: ultralytics/engine/results.py [hunk 1]
```diff
@@ -259,6 +260,7 @@ def __init__(
         self.probs = Probs(probs) if probs is not None else None
         self.keypoints = Keypoints(keypoints, self.orig_shape) if keypoints is not None else None
         self.obb = OBB(obb, self.orig_shape) if obb is not None else None
+        self.semseg = semseg  # auxiliary semantic segmentation logits (C, H, W) or None
```
Comment (UltralyticsAssistant) [inline ultralytics/engine/results.py:263]: ⚠️ **HIGH**: `semseg` is stored on `Results`, but it is not integrated into the class' tensor propagation helpers. As written, common operations like `result.cpu()`, `result.cuda()`, `result.to()`, and `result.numpy()` will leave `semseg` behind on its original device/type, and any copy/new helpers may silently drop it. That makes the new field inconsistent with the rest of the result payload in downstream inference code.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The semseg inference plumbing looks coherent overall, and the keypoint flip guard is a good edge-case fix. The main issue I see is that `Results.semseg` is only attached at construction time; without updating the `Results` tensor/copy machinery, it will not behave like the other tensor-backed fields and can easily become stale or be lost in downstream usage.

💬 Posted 1 inline comment

**Comment by glenn-jocher** [comment]
Please keep this PR scoped to the semantic inference change only, move the unrelated `v8_transforms()` and DDP edits into separate PRs, and if possible align the output with the existing [`semantic_mask` result API](https://docs.ultralytics.com/modes/predict) and [semantic predictor flow](https://docs.ultralytics.com/reference/models/yolo/semantic/predict) instead of adding a separate `semseg` field; a small test and docs update would also help review.

**Comment by glenn-jocher** [comment]
Thank you @tedo001 for putting this together and for working through the semantic-output plumbing. The underlying capability is valuable, and your contribution helped clarify the result propagation requirements. Since `main` now has the supported `semantic_mask` architecture across Results conversion/copy/plotting and the current branch also carries unrelated augmentation/DDP changes, we’re closing this PR as superseded rather than asking you to reconcile the parallel `semseg` API. We genuinely appreciate the work and hope to see you contribute again—focused improvements to the current `semantic_mask` path would be very welcome. 🙏
